### ISSUE TITLE
Bug Report: Expected YAML collection at dev_dependencies. 

### ISSUE THREAD

**Issue by simphotonics** [jeroen-meijer/pubspec-assist#49]
Hello @jeroen-meijer, thank you for creating this very helpful package.

The above issue has been reported already but I'd like to add platform details to help reproduce the error:
```Console
$ dart --version
Dart SDK version: 2.9.0-21.10.beta (beta) (Tue Jul 21 10:46:30 2020 +0200) on "linux_x64"
``` 

The error occurs when the `dependencies` or `dev_dependencies` entries are initially empty:
```Yaml
name: quote_buffer
description:
  Extension on StringBuffer that simplifies transforming single objects and lists of objects
  into quoted strings.


version: 0.2.0
homepage: https://github.com/simphotonics/quote_buffer

environment:
  sdk: '>=2.9.0 <3.0.0'

dependencies:

dev_dependencies:
```  

When selecting any package, e.g. pedantic, the following error message is shown:
`Pubspec Assist: A critical error has occurred. If this happens again, please report it.
 Error message: Expected YAML collection at dev_dependencies. Remaining path: pedantic`

**Comment by jeroen-meijer** [comment]
Hi there,

Thank you for the nice works, and for opening the issue!

I'll have a look at the issue soon.

That codesample you attached, is that the entire pubspec file at the time you try to add the dependency?

**Comment by simphotonics** [comment]
@jeroen-meijer, thanks for taking the time to look into this. Yes, I included the entire content of the `pubspec` file.

I just reproduced the issue using the lastest version of vscode:
```
Version: 1.49.3
Commit: 2af051012b66169dde0c4dfae3f5ef48f787ff69
Date: 2020-10-02T17:56:11.027Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 5.4.0-48-generic
```  

Steps to reproduce the error:
1) pubspec file with empty `dependencies`. 
2) press F1 in vscode and select `Pubspec Assist: Add/update dependency`
3) Enter a package name  and confirm.

Similarly: 
1) pubspec file with empty `dev_dependencies`. 
2) press F1 in vscode and select `Pubspec Assist: Add/update dev dependency`
3) Enter a package name  and confirm.
